src/python/py/models/builder.py

```diff
@@ -166,15 +166,15 @@
                 "do_sample": config.do_sample if hasattr(config, "do_sample") else False,
                 "early_stopping": True,
                 "length_penalty": config.length_penalty if hasattr(config, "length_penalty") else 1.0,
-                "max_length": config.max_length if hasattr(config, "max_length") else 20,
+                "max_length": self.context_length,
```

Comment:
>self.context_length, [](http://example.com/codeflow?start=30&length=20)

is it better to use:
"max_length": config.max_length if hasattr(config, "max_length") else self.context_length,

Reply:
The `config` object appears to always have the `max_length` attribute so the `else` case is never used. If `max_length` is in the original JSON config, then `config.max_length = json_value`. If `max_length` is not in the original JSON config, then `config.max_length = 20` by default. Because the latter case is more frequent, `max_length` is often set to 20 in the GenAI config.

If `max_length` is in the original JSON config, it is typically set to the context length (e.g. in LLaMA-2 7B [here](https://huggingface.co/meta-llama/Llama-2-7b-hf/blob/8cca527612d856d7d32bd94f8103728d614eb852/generation_config.json#L7)). Thus, it is easier to set `max_length = context_length` as the default in the GenAI config. Users can then manually change the `max_length` as needed.